pkg/metrics/record.go

```diff
@@ -76,9 +76,12 @@ func RecordParserDuration(ctx context.Context, trigger, source, status string, s
 }
 
 // RecordLastSync produces a measurement for the LastSync view.
-func RecordLastSync(ctx context.Context, commit string, timestamp time.Time) {
+func RecordLastSync(ctx context.Context, status, commit string, timestamp time.Time) {
+	tagCtx, _ := tag.New(ctx,
+		tag.Upsert(KeyStatus, status),
+		tag.Upsert(KeyCommit, commit))
```

Comment:
Just want to confirm the commit tag will not increase the cardinality?

Reply:
It will, but no more than already present in the last_apply_timestamp, which has the same tags.

My hope is that the resource attributes added by https://github.com/GoogleContainerTools/kpt-config-sync/pull/100 and https://github.com/GoogleContainerTools/kpt-config-sync/pull/114 will mitigate the issue so that it's only a problem if you have 5k+ commits in the same repo PER RECONCILER POD.

And if that becomes a problem, you can simply delete the reconciler pod and re-create it to start over on a new time series with new resource attributes.

More likely tho, a node replacement or upgrade will cause replacement first.

Reply:
Good to know.

Reply:
Also, the 5k+ limit is a recommendation is for Monarch. Cloud Monitoring actually documents 30k as the limit: https://cloud.google.com/logging/docs/logs-based-metrics/troubleshooting#too-many-time-series